vcr/cassette.py

```diff
@@ -337,10 +344,29 @@ def find_requests_with_most_matches(self, request):
 
         return final_best_matches
 
+    def _new_interactions(self):
+        """List of new HTTP interactions (request/response tuples)"""
+        new_interactions = []
+        for request, response in self.data:
+            if all(
+                not requests_match(request, old_request, self._match_on)
+                for old_request, _ in self._old_interactions
+            ):
+                new_interactions.append((request, response))
+        return new_interactions
+
     def _as_dict(self):
-        return {"requests": self.requests, "responses": self.responses}
+        requests = self.requests
+        responses = self.responses
+        if self.drop_unused_requests:
+            interactions = self._played_interactions + self._new_interactions()
+            requests = [request for request, _ in interactions]
+            responses = [response for _, response in interactions]
+        return {"requests": requests, "responses": responses}
 
     def _save(self, force=False):
+        if len(self._played_interactions) < len(self._old_interactions):
+            force = True
```

Comment:
Is this addition missing a check for `drop_unused_requests`?

Reply:
You're right, it does need a check for drop_unused_requests.